### PR TITLE
[doc] Update dartlang.org links to dart.dev

### PR DESCRIPTION
**Pull request by ayan-b**
Also update dartbug.com to github issue tracker.

### CODE REVIEW

**Review by kevmoo — COMMENTED** [review]
A couple of tweaks...

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -11,14 +11,14 @@ If you'd like to improve the documentation, you have three options:
...
 
-Before you start working on a larger contribution, you should get in touch with us first through the  [Dart Issue Tracker](http://dartbug.com) with your idea so that we can help out and possibly guide you. Coordinating up front makes it much easier to avoid frustration later on.
+Before you start working on a larger contribution, you should get in touch with us first through the  [Dart Issue Tracker](https://github.com/dart-lang/sdk/issues) with your idea so that we can help out and possibly guide you. Coordinating up front makes it much easier to avoid frustration later on.
```
Comment (kevmoo) [inline CONTRIBUTING.md:21]: https://dartbug.com is fine. Although change the URI to https

Comment (ayan-b) [reply CONTRIBUTING.md]: Reverted the change.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -11,14 +11,14 @@ If you'd like to improve the documentation, you have three options:
...
-  * Improve the API reference docs at [api.dartlang.org](https://api.dartlang.org) by editing doc comments in the [Dart SDK repo](https://github.com/dart-lang/sdk/tree/master/sdk/lib). For more information on how to write API docs, see [Effective Dart: Documentation](https://www.dartlang.org/guides/language/effective-dart/documentation).
+  * Contribute to the Dart developer websites such as [www.dart.dev](https://www.dart.dev) (repo: [dart-lang/site-www](https://github.com/dart-lang/site-www)) and [dart.dev/web](https://dart.dev/web) (repo: [dart-lang/site-webdev](https://github.com/dart-lang/site-webdev)). For more information, see [Writing for Dart and Flutter websites](https://github.com/dart-lang/site-shared/wiki/Writing-for-Dart-and-Flutter-websites).
```
[the remark this answers is not in the document]
Comment (ayan-b) [reply CONTRIBUTING.md]: Dropped `www`.

**Comment by kevmoo** [comment]
See https://dart-review.googlesource.com/c/sdk/+/142640

**Comment by ayan-b** [comment]
Thanks!

**Pull request closed without merging**